File: backend/internal/model/dependent_service.go
package model

import "time"

type DependentService struct {
	ID                  uint             `gorm:"primaryKey" json:"id"`
	ServiceCode         string           `gorm:"uniqueIndex;size:80;not null" json:"service_code"`
	Name                string           `gorm:"size:160;not null" json:"name"`
	OwnerTeam           string           `gorm:"size:160;not null;index" json:"owner_team"`
	Environment         string           `gorm:"size:40;not null;index" json:"environment"`
	ChainID             uint             `gorm:"not null;index" json:"chain_id"`
	Chain               CertificateChain `gorm:"constraint:OnUpdate:CASCADE,OnDelete:RESTRICT" json:"chain"`
	ClientTrustRefsJSON string           `gorm:"type:text;not null" json:"client_trust_refs_json"`
	Protocol            string           `gorm:"size:40;not null" json:"protocol"`
	Criticality         string           `gorm:"size:24;not null;index" json:"criticality"`
	DependencyEdgesJSON string           `gorm:"type:text;not null" json:"dependency_edges_json"`
	ServiceState        string           `gorm:"size:24;not null;index" json:"service_state"`
	CreatedAt           time.Time        `gorm:"not null" json:"created_at"`
	UpdatedAt           time.Time        `gorm:"not null" json:"updated_at"`
}

File: frontend/src/types/dependent-service.ts
import type { CertificateChain } from './certificate-chain'

export type ServiceState = 'active' | 'inactive'
export type Criticality = 'low' | 'medium' | 'high' | 'critical'
export type Environment = 'development' | 'staging' | 'production' | 'disaster_recovery'
export type Protocol = 'tls' | 'mtls' | 'ldaps' | 'smtps' | 'kafka_tls'

export interface DependentService {
  id: number
  service_code: string
  name: string
  owner_team: string
  environment: Environment
  chain_id: number
  chain?: CertificateChain
  client_trust_refs_json: number[]
  protocol: Protocol
  criticality: Criticality
  dependency_edges_json: number[]
  service_state: ServiceState
  created_at: string
  updated_at: string
}

export interface DependentServiceInput {
  service_code?: string
  name: string
  owner_team: string
  environment: Environment
  chain_id: number
  client_trust_refs_json: number[]
  protocol: Protocol
  criticality: Criticality
  dependency_edges_json: number[]
}

